cg/apps/deliverables_metrics_parser/parser/deliverables_parser.py

```diff
@@ -0,0 +1,122 @@
+"""Module to parse the metrics deliverables file."""
+import operator
+from enum import Enum
+from functools import reduce
+from pathlib import Path
+
+from cg.apps.deliverables_metrics_parser.models.pipeline_metrics_deliverables import (
+    MIPDNAMetricsDeliverables,
+)
+from cg.constants.constants import FileFormat
+from cg.constants.pipeline import Pipeline
+from cg.io.controller import ReadFile
+
+
+class MetricsParserConstants(Enum):
+    NAME: str = "name"
+    VALUE: str = "value"
+    ID: str = "id"
+    METRICS: str = "metrics"
+
+
+# we need some kind of standardisation across pipelines for these fields
+READ_INFORMATION_FIELDS = [
+    "reads_mapped",
+    "raw_total_sequences",
+    "percentage_mapped_reads",
+    "PCT_PF_READS_IMPROPER_PAIRS",
+]
+
+
+def generate_metrics_deliverables_model(
+    sample_id: str, sample_metrics: dict
+) -> MIPDNAMetricsDeliverables:
+    """Generate the metrics deliverable model for a sample."""
+    sample_metrics["id"] = sample_id
+    return MIPDNAMetricsDeliverables.model_validate(sample_metrics)
+
+
+def get_metrics_deliverables_file_path(pipeline: str, case_id: str) -> Path:
```

Comment:
I need to think of a way to improve the pipelines dict suggestions welcome

Reply:
the `IntermediateDirectories` could be in constants

Reply:
I like this, although the `path_root` and `file_suffix` should be (and the root might already be) in constants.

Reply:
It woukd not work like this, we are missing the "proper" pipeline level for some pipelines. The folder for "mip-dna" and "mip-rna" is set to "rare-disease" for example... so there is on direct way of doing this. @karlnyr mentioned a pipeline root dir in the config. will have a look and try with that.